docs/anatomy.md

```diff
@@ -0,0 +1,23 @@
+# Anatomy of a deep learning library
```

Comment:
Could you please convert this to asciidoc format (extension `.adoc`), since we'll be using that for all the docs.